text_extensions_for_pandas/array/tensor.py

```diff
@@ -224,14 +224,24 @@ def take(
             # `indices` indicate missing values and are set to `fill_value`
             indices = np.asarray(indices, dtype=np.intp)
             validate_indices(indices, len(self._tensor))
-            if fill_value is None:
-                fill_value = np.nan
-            values = np.full((len(indices),) + self._tensor.shape[1:], fill_value)
-            for i, idx in enumerate(indices):
-                if idx >= 0:
-                    values[i] = self._tensor[idx]
-        else:
-            values = self._tensor.take(indices, axis=0)
+
+            # Check if there are missing indices to fill, if not can use numpy take below
+            has_missing = np.any(indices < 0)
+            if has_missing:
```

Comment:
fyi, need this check so that it doesn't unnecessarily promote ints to floats if there are no missing values. Noticed this when sorting a DataFrame with TensorArray, which is now in a cell of the notebook in https://github.com/CODAIT/text-extensions-for-pandas/pull/122